### PR TITLE
MueLu: Pushing block ids down the stack

### PR DESCRIPTION
**Pull request by csiefer2**
TESTS: RHEL7

### CODE REVIEW

File: packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_def.hpp [hunk 1]
```diff
@@ -132,80 +133,90 @@ namespace MueLu {
...
     if (pL.get<bool>("lightweight wrap") == true) {
-      if (pL.get<std::string>("aggregation: drop scheme") == "distance laplacian")
-        Input(currentLevel, "Coordinates");
-
-    }
+      std::string algo = pL.get<std::string>("aggregation: drop scheme");
+      if (algo == "distance laplacian" || algo == "block diagonal distance laplacian") {
+        Input(currentLevel, "Coordinates");    
+      }
+      else if (algo == "block diagonal classical" || algo == "block diagonal distance laplacian" || algo == "block diagonal")  {
```
Comment (cgcgcg) [inline packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_def.hpp:155]: This looks a bit odd to me, "block diagonal distance laplacian" is both in `if` and `else if`

Comment (csiefer2) [reply packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_def.hpp:155]: Yeah, that else isn't right

Comment (csiefer2) [reply packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_def.hpp]: Fixed

**Review by cgcgcg — APPROVED** [review]

**Comment by csiefer2** [comment]
@trilinos/framework  GCC 8.3.0 build is completely busted:

--------------------------------------------------------------------------
Open MPI was unable to obtain the username in order to create a path
for its required temporary directories.  This type of error is usually
caused by a transient failure of network-based authentication services
(e.g., LDAP or NIS failure due to network congestion), but can also be
an indication of system misconfiguration.

Please consult your system administrator about these issues and try
again.
--------------------------------------------------------------------------
[ascic141:02107] 2 more processes have sent help message help-orte-runtime.txt / orte:session:dir:nopwname
[ascic141:02107] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages

**Comment by csiefer2** [comment]
@trilinos/framework  Wow.  half the autotester jobs didn't even *run*

Y'all should fix this

**Pull request merged**